File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        #Dictionary Solution
        #Time: O(n) Space: O(n)
        
        table = {v:i for i,v in enumerate(nums)}
        for i,v in enumerate(nums):
            if target-v in table and table[target-v] != i:
                return [i,table[target-v]]
        return -1 #error

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
         
        //Convert array to hashtable
        //Iterate through array and see if matching value is in hashtable
        // If value in hashtable, return index pair
        
        HashMap<Integer, Integer> hash = new HashMap<Integer, Integer>();
        int n = nums.length;
        for (int i = 0; i < n; i++) {
            hash.put(nums[i], i); 
        }
        
        for (int i = 0; i < n; i++) {
            int val = target - nums[i];
            if (hash.containsKey(val) && hash.get(val) != i) {
                return new int[]{i, hash.get(val)};
            } 
        }
        
        return new int[]{};
    }
}